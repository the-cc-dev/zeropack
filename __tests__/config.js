const { getDefaultOptions, getFlowOptions, getWebpackOptions } = require("..");

test("default", async () => {
  expect(await getDefaultOptions()).toMatchSnapshot();
});

test("flow", async () => {
  expect(await getFlowOptions()).toMatchSnapshot();
});

test("webpack", async () => {
  expect(await getWebpackOptions()).toMatchSnapshot();
});
