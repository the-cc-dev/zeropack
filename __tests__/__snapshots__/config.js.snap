// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default 1`] = `
Object {
  "devDependencies": Object {},
  "main": "dist/index.js",
  "name": "Unknown",
  "zeropack": Object {
    "externals": Array [],
    "mains": Array [
      "browser",
      "main",
      "module",
    ],
    "mode": "development",
    "name": "Unknown",
    "src": "./src/index.js",
  },
}
`;

exports[`flow 1`] = `Array []`;

exports[`webpack 1`] = `
Array [
  Object {
    "devtool": "source-map",
    "entry": "./src/index.js",
    "externals": Array [],
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "test": /\\\\\\.jsx\\?\\$/,
          "use": Array [
            Object {
              "loader": "babel-loader",
              "options": Object {
                "presets": Array [
                  Array [
                    Object {
                      "__esModule": true,
                      "default": [Function],
                      "isPluginRequired": [Function],
                      "transformIncludesAndExcludes": [Function],
                    },
                    Object {
                      "targets": Object {
                        "node": Promise {},
                      },
                    },
                  ],
                  Object {
                    "env": Object {
                      "development": Object {
                        "plugins": Array [],
                      },
                    },
                    "plugins": Array [
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "presets": Array [
                      Object {
                        "plugins": Array [
                          [Function],
                        ],
                      },
                    ],
                  },
                  Object {
                    "plugins": Array [
                      [Function],
                      [Function],
                    ],
                    "presets": Array [
                      Object {
                        "plugins": Array [
                          [Function],
                          [Function],
                        ],
                        "presets": Array [
                          Object {
                            "plugins": Array [
                              [Function],
                              [Function],
                              [Function],
                            ],
                            "presets": Array [
                              Object {
                                "plugins": Array [
                                  [Function],
                                  [Function],
                                  [Function],
                                  [Function],
                                  [Function],
                                ],
                              },
                            ],
                          },
                        ],
                      },
                    ],
                  },
                ],
                "sourceMaps": true,
              },
            },
          ],
        },
      ],
    },
    "output": Object {
      "filename": "index.js",
      "library": "zeropack",
      "libraryTarget": "umd",
      "path": "/Users/tshugart/Sites/treshugart/zeropack/dist",
    },
  },
]
`;
